### ISSUE TITLE
Doesn't build on Arch (it used to)

### ISSUE THREAD

**Issue by raingloom** [Harvey-OS/harvey#644]
# Steps to reproduce
use an up-to-date Arch Linux to build the master branch

# Compiler output
compiled with: `CC=gcc ARCH=amd64 util/build`

and: `CC=clang ARCH=amd64 util/build`

both gave the same error

```
9660srv.o: In function `iattach':
/home/rain/VMs/Harveydev/harvey/sys/src/cmd/9660srv/9660srv.c:212: undefined reference to `__stack_chk_fail'
9660srv.o: In function `iwalkup':
/home/rain/VMs/Harveydev/harvey/sys/src/cmd/9660srv/9660srv.c:249: undefined reference to `__stack_chk_fail'
9660srv.o: In function `iwalk':
/home/rain/VMs/Harveydev/harvey/sys/src/cmd/9660srv/9660srv.c:306: undefined reference to `__stack_chk_fail'
9660srv.o: In function `ireaddir':
/home/rain/VMs/Harveydev/harvey/sys/src/cmd/9660srv/9660srv.c:360: undefined reference to `__stack_chk_fail'
9660srv.o: In function `opendotdot':
/home/rain/VMs/Harveydev/harvey/sys/src/cmd/9660srv/9660srv.c:565: undefined reference to `__stack_chk_fail'
9660srv.o:/home/rain/VMs/Harveydev/harvey/sys/src/cmd/9660srv/9660srv.c:755: more undefined references to `__stack_chk_fail' follow
2017/09/08 00:02:43 exit status 1
  71.02s user 19.46s system 65% cpu 2:17.34 total
```

**Comment by elbing** [comment]
That symbol looks like is not in harvey's code. Please, could you post the linking line previous to error? Also the version of binutils and gcc. Thanks.

**Comment by rminnich** [comment]
This is the stack protector crap.
We have to find the switch to turn it off.

**Comment by rminnich** [comment]
The switch to fix this is -fno-stack-protector

what clang will do with that is anybody's guess.

**Comment by elbing** [comment]
Solved in #645
